pretyping/detyping.ml

```diff
@@ -476,7 +476,7 @@ let rec detype flags avoid env sigma t = CAst.make @@
  	| _ -> GApp (f',args')
       in
       mkapp (detype flags avoid env sigma f)
-        (detype_array flags avoid env sigma args)
+        (Array.map_to_list (detype flags avoid env sigma) args)
```

Comment:
This changes the order in which elements of `args` are detyped, is that likely to be an issue?